pkg/api/uiinfo/ui_info_handler.go

```diff
@@ -0,0 +1,38 @@
+package uiinfo
+
+import (
+	"net/http"
+	"os"
+
+	"github.com/harvester/harvester/pkg/config"
+	ctlharvesterv1 "github.com/harvester/harvester/pkg/generated/controllers/harvesterhci.io/v1beta1"
+	"github.com/harvester/harvester/pkg/settings"
+	"github.com/harvester/harvester/pkg/util"
+)
+
+type Handler struct {
+	settingsCache ctlharvesterv1.SettingCache
+}
+
+func NewUIInfoHandler(scaled *config.Scaled, option config.Options) *Handler {
+	return &Handler{
+		settingsCache: scaled.HarvesterFactory.Harvesterhci().V1beta1().Setting().Cache(),
+	}
+}
+
+func (h *Handler) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
+	uiSource := settings.UISource.Get()
+	if uiSource == "auto" {
+		if !settings.IsRelease() {
+			uiSource = "external"
+		} else {
+			uiSource = "bundled"
+		}
+	}
+	util.ResponseOKWithBody(rw, map[string]string{
+		settings.UISourceSettingName:               uiSource,
+		settings.UIIndexSettingName:                settings.UIIndex.Get(),
+		settings.UIPluginIndexSettingName:          settings.UIPluginIndex.Get(),
+		settings.UIPluginBundledVersionSettingName: os.Getenv(settings.GetEnvKey(settings.UIPluginBundledVersionSettingName)),
```

Comment:
wondering what this setting response is used for?

Reply:
cc @WuJun2016

Reply:
The front-end needs to know what version is saved inside plugin-assets.  I know that normally the version of the plugin is the same as the harvester version, so its value should be the same as the `server-version`, but I'm not sure if using the `server-version` value is always appropriate.    
So to make sure I can always get the right value, so I added a property to save the version of the plugin (this plugin version is determined by the name of the packaged file)

If you don't add this attribute, then you need to expose the `server-version` to ui-info.

Reply:
I tested it, the `server-version` user is able to change, which is actually unreasonable.   
if the user changes this value, the user will not be able to access the Harvester from multiple clusters